tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/FileHelper.cs

```diff
@@ -38,5 +55,695 @@ public static class FileHelper
+        /// <param name="relativeTo">Base path used to compute relative output paths and as the root for glob evaluation.</param>
+        /// <param name="totalBudget">Maximum aggregate character budget for loaded content.</param>
+        /// <param name="perFileLimit">Maximum per-file character allocation before truncation occurs.</param>
+        /// <param name="priorityFunc">Function computing file priority (lower loads first). Should be inexpensive; called once per candidate.</param>
+        /// <param name="logger">Optional logger for discovery and planning diagnostics.</param>
+        /// <param name="ct">Cancellation token applied only during file reads.</param>
+        /// <returns>A concatenated representation of included files in XML-ish blocks with metadata attributes.</returns>
+        /// <remarks>
+        /// Behaves identically to the <see cref="LoadFilesAsync(IEnumerable{SourceInput}, string, int, int, Func{FileMetadata, int}, ILogger?, CancellationToken)"/> overload except that filtering rules are shared across all <paramref name="filePaths"/>.
+        /// </remarks>
+        public static async Task<string> LoadFilesAsync(
+            IEnumerable<string> filePaths,
+            string[] includeExtensions,
+            string[] excludeGlobPatterns,
+            string relativeTo,
+            int totalBudget,
+            int perFileLimit,
+            Func<FileMetadata, int> priorityFunc,
+            ILogger? logger = null,
+            CancellationToken ct = default)
+        {
+            var plan = CreateFileLoadingPlan(filePaths, includeExtensions, excludeGlobPatterns, relativeTo, totalBudget, perFileLimit, priorityFunc, logger);
+            return await ExecuteFileLoadingPlanAsync(plan, logger, ct);
+        }
+
+        /// <summary>
+        /// Convenience overload for loading files from a single directory using shared filtering rules.
+        /// </summary>
+        /// <param name="dir">Directory to scan recursively.</param>
+        /// <param name="includeExtensions">Extensions to include or empty array for all.</param>
+        /// <param name="excludeGlobPatterns">Glob exclusion patterns relative to <paramref name="relativeTo"/>.</param>
+        /// <param name="relativeTo">Root path for relative output formatting.</param>
+        /// <param name="totalBudget">Aggregate character budget across all files.</param>
+        /// <param name="perFileLimit">Maximum characters per file before truncation.</param>
+        /// <param name="priorityFunc">Priority selector (lower first); see remarks in other overloads.</param>
+        /// <param name="logger">Optional logger.</param>
+        /// <param name="ct">Cancellation token applied during file reads only.</param>
+        /// <returns>Concatenated structured content string for included files.</returns>
+        public static async Task<string> LoadFilesAsync(
```

Comment:
Overload is unused

Reply:
I prefer to keep these overloads. This should be the go-to API for loading files across all tools and I expect these overloads to be valuable in many scenarios.